src/wallet/scriptpubkeyman.cpp

```diff
@@ -122,7 +164,36 @@ bool LegacyScriptPubKeyMan::CanGetAddresses(bool internal)
 
 bool LegacyScriptPubKeyMan::Upgrade(int prev_version, int new_version, std::string& error)
```

Comment:
IIRC I had that there originally for upgrading a legacy wallet to a descriptor wallet. But since we are no longer doing that, it can be removed. I believe it was going to be used in combination with #15761 (or rather #15761 was split from an earlier implementation).

Reply:
Removed `new_version`.

For anyone who is wondering why this is works and/or is safe, the new version is set as the wallet maximum version so all of the `CanSupportFeature` calls will fail if the specified new version is too low.